[ASPA-008] Recording User Information to Gsheet

Ray-F opened the pull request:
**Issue:** 
Needed to add read/write functions in Gsheet_Interface_Model to provide an interface for function calls from outside Model class.

**Solution:** 
Added 3 main functions (record_to_sheet(*args), get_from_sheet(*args), get_sheet_size()) to perform all necessary read/write functionality required under the scope of this project.

**Risk area:** 
Should be self-contained and functions present no risk themselves. Errors in function usage may be present (no error catching for function arg input errors)

**Reviewed by**
Jiaru, Lucas

Commit:
[ASPA-011] Adding a random file for testing purposes
Issue: Testing git
Solution: Made a file
Risk Area: n/a
Reviewed by: Emily

Commit:
[ASPA-011] Added google sheets interface linkage
Issue: Linking sheets API
Solution: Done (put in models)
Risk Area: semi
Reviewed by: Lucas

lucas2005gao committed:
"Issue: Refactoring causing failures in SheetsAPI
Solution: Found typo and changed composer requirements
Risk Area: none if no calls to Sheets API
Reviewed by: Lucas"

Commit:
[ASPA-008 & ASPA-011] SheetsAPI streamline and tidy-up
Issue: Gsheets Model messy
Solution: Added functions call to Gsheets Model, with comments
Risk: Low risk – make sure to use new functions record_to_sheet(*args) and get_from_sheet(*args) for any sheets usage.
Reviewed by: Lucas / Jiaru

Commit:
[ASPA-008 & ASPA-011] SheetsAPI streamline and tidy-up
Issue: Gsheets Model messy
Solution: Added functions call to Gsheets Model, with comments
Risk: Low risk – make sure to use new functions record_to_sheet(*args) and get_from_sheet(*args) for any sheets usage.
Reviewed by: Lucas / Jiaru

Commit:
[ASPA-008] Added recording user functionality, and tidied up gsheets model
Issue: User recording
Solution: Added function record_to_sheet(*args) and function get_from_sheet(*args)
Risk: Could collide with other works on Gsheet Model?
Reviewed by: Lucas Gao

Commit:
[ASPA-008] Adding comments

Commit:
[ASPA-008] Removed conflict messages from git

Commit:
[ASPA-008] Removed need for UPI and Student ID
Risk: none

Commit:
[ASPA-008] Made changes to way sheet information is read
Issue: Could only read from one cell w/ one function call
Solution: Made so a range selection (e.g A1:B5) can be made, NB: returns an array as commented
Risk: none – no one has used this function call yet so no changes needed.
Reviewed by: -

Commit:
[ASPA-008] Added example read request under testing block

lucas2005gao reviewed (commented):
Well done Raymond! Just a few minor things, can we have the spreadsheetID as a constant inside the application->config->constants.php file, and can we have the color code changed to red=0.7, green=0.9, blue=0.7 for the highlighting colour please.

Commit:
[ASPA-008] Changed spreadsheet ID to be a constant, found in constants.php (in config folder)
Issue: Spreadsheet ID needs to be editable from constants file
Solution: Changed location of spreadsheet ID definition to be in constants file
Risk: -
Reviewed by: Lucas

Commit:
[ASPA-008] Changed highlighting colour from yellow to a green shade (per direction of Lucas)
...
Reviewed by: Lucas

Ray-F commented:
> Well done Raymond! Just a few minor things, can we have the spreadsheetID as a constant inside the application->config->constants.php file, and can we have the color code changed to red=0.7, green=0.9, blue=0.7 for the highlighting colour please.

All done!

lucas2005gao reviewed (approved):
Looks great, good job Raymond!
